textx/model_params.py

```diff
@@ -0,0 +1,98 @@
+"""
+Management of parameters passed to model_from_str or model_from_file.
+"""
+
+import sys
+from functools import reduce
+from collections import namedtuple
+from textx.exceptions import TextXError
+
+if sys.version < '3':
+    from collections import Mapping
+else:
+    from collections.abc import Mapping
+
+
+class ModelParams(Mapping):
+    """A read only dictionary that protocols
+    accessing the values.
+
+    This way, it is possible to check after parsing a model
+    if all parameters have been used by accessing the value
+    directly.
+
+    https://stackoverflow.com/questions/3387691/how-to-perfectly-override-a-dict
+    https://docs.python.org/3/library/collections.abc.html
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.store = dict(*args, **kwargs)
+        self.used_keys = set()
+
+    def __getitem__(self, key):
+        self.used_keys.add(key)
+        return self.store[self.__keytransform__(key)]
+
+    def __iter__(self):
+        return iter(self.store)
+
+    def __len__(self):
+        return len(self.store)
+
+    def __keytransform__(self, key):
+        return key
+
+    def _have_all_parameters_been_used(self):
+        return reduce(
+            lambda r, k: r and (k in self.used_keys),
+            self.store.keys(), True)
+
+    all_been_used = property(
+        lambda self: self._have_all_parameters_been_used(),
+        doc="returns if all parameters have been used by the meta model")
```

Comment:
This part of code (property def and previous method) can be replaced with:
```python
        @property
        def all_been_used(self):
            "returns if all parameters have been used by the meta model"
            return not (set(self.store.keys()) - set(self.used_keys))
```

Another shortening suggestion is making this property called `all_used`.  It is shorter to read/type while not being less understandable

Reply:
done.